group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorRuntimeMetrics.java

```diff
@@ -0,0 +1,266 @@
+     */
+    private final Metrics metrics;
+
+    /**
+     * The partition load sensor.
+     */
+    private Sensor partitionLoadSensor;
+
+    /**
+     * The thread idle sensor.
+     */
+    private Sensor threadIdleRatioSensor;
+
+    /**
+     * The event queue time updater.
+     */
+    private final Consumer<Long> eventQueueTimeUpdater;
+
+    /**
+     * The event queue processing time updater.
+     */
+    private final Consumer<Long> eventQueueProcessingTimeUpdater;
+
+    public GroupCoordinatorRuntimeMetrics(MetricsRegistry registry, Metrics metrics) {
+        this.registry = Objects.requireNonNull(registry);
+        this.metrics = Objects.requireNonNull(metrics);
+
+        this.numPartitionsLoading = getMetricName("num-partitions-loading");
+        this.numPartitionsActive = getMetricName("num-partitions-active");
+        this.numPartitionsFailed = getMetricName("num-partitions-failed");
+        this.eventQueueSize = getMetricName("event-queue-size");
+
+        initializeMetrics();
+
+        eventQueueTimeUpdater = newHistogram(eventQueueTimeMs);
+        eventQueueProcessingTimeUpdater = newHistogram(eventQueueProcessingTimeMs);
+    }
+
+    /**
+     * Register the metrics.
+     */
+    public void initializeMetrics() {
+        metrics.addMetric(numPartitionsLoading, (Gauge<Long>) (config, now) -> numPartitionsLoadingCounter.get());
+        metrics.addMetric(numPartitionsActive, (Gauge<Long>) (config, now) -> numPartitionsActiveCounter.get());
+        metrics.addMetric(numPartitionsFailed, (Gauge<Long>) (config, now) -> numPartitionsFailedCounter.get());
+
+        this.partitionLoadSensor = metrics.sensor("GroupPartitionLoadTime");
+        this.partitionLoadSensor.add(
+            metrics.metricName(
+                "partition-load-time-max",
+                METRICS_GROUP,
+                "The max time it took to load the partitions in the last 30 sec."
+            ), new Max());
+        this.partitionLoadSensor.add(
+            metrics.metricName(
+                "partition-load-time-avg",
+                METRICS_GROUP,
+                "The average time it took to load the partitions in the last 30 sec."
+            ), new Avg());
+
+        this.threadIdleRatioSensor = metrics.sensor("ThreadIdleRatio");
+        this.threadIdleRatioSensor.add(
+            metrics.metricName(
+                "thread-idle-ratio-min",
+                METRICS_GROUP,
+                "The minimum thread idle ratio over the last 30 seconds."
+            ), new Min());
+        this.threadIdleRatioSensor.add(
+            metrics.metricName(
+                "thread-idle-ratio-avg",
+                METRICS_GROUP,
+                "The average thread idle ratio over the last 30 seconds."
+            ), new Avg());
+    }
+
+    /**
+     * Retrieve the yammer metric name.
+     * @param name The name of the metric.
+     *
+     * @return The metric name.
+     */
+    private MetricName getMetricName(String name) {
```

Comment:
This would be the non-yammer metric name right? since the return type is not `com.yammer.metrics.core.MetricName`

Reply:
That's correct. I updated both GroupCoordinatorRuntimeMetrics and GroupCoordinatorRuntimeMetricsTest to always use `com.yammer.metrics.core.MetricName` for yammer metrics and `MetricName` for kafka metrics.

Also, i fixed the comments and renamed this to kafkaMetricName